core/src/main/java/org/frankframework/receivers/FrankListener.java

```diff
@@ -0,0 +1,147 @@
+/*
+   Copyright 2024 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package org.frankframework.receivers;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.lang3.StringUtils;
+import org.frankframework.configuration.Configuration;
+import org.frankframework.configuration.ConfigurationException;
+import org.frankframework.core.Adapter;
+import org.frankframework.core.HasPhysicalDestination;
+import org.frankframework.core.IMessageHandler;
+import org.frankframework.core.IPushingListener;
+import org.frankframework.core.IbisExceptionListener;
+import org.frankframework.core.ListenerException;
+import org.frankframework.core.PipeLineResult;
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.doc.Category;
+import org.frankframework.stream.Message;
+import org.springframework.context.ApplicationContext;
+
+
+/**
+ * Listener to accompany the {@link org.frankframework.senders.FrankSender}, for situations where
+ * calling an {@link Adapter} directly is not desired. This could be because message / error logging is required for messages
+ * sent to the subadapter.
+ * <br/>
+ * See the {@link org.frankframework.senders.FrankSender} documentation for more information.
+ */
+@Category("Basic")
+@Log4j2
+public class FrankListener implements IPushingListener<Message>, HasPhysicalDestination, ServiceClient {
+
+	private static final ConcurrentMap<String, FrankListener> listeners = new ConcurrentHashMap<>();
+
+	private final @Getter String domain = "JVM";
+	private @Getter @Setter ApplicationContext applicationContext;
+	private final @Getter ClassLoader configurationClassLoader = Thread.currentThread().getContextClassLoader();
+
+	private @Getter String name;
+
+	private @Getter boolean open=false;
+	private @Getter @Setter IMessageHandler<Message> handler;
+
+	public static @Nullable FrankListener getListener(String name) {
+		return listeners.get(name);
+	}
+
+	@Override
+	public String getPhysicalDestinationName() {
+		return getConfiguration().getName() + "/" + getName();
```

Comment:
Je zou hier ook applicationContext.getId() kunnen gebruiken, maar dit is misschien voor de ontwikkelaars iets handiger :)

Reply:
Deze kan meerdere malen opgehaald worden maar hij veranderd nooit.
Misschien dat je dit in een configure stap kan berekenen en dan kan gebruiken bij `getPhysicalDestinationName()` alsook `fullName`.